File: app/src/main/java/team/singularity/scoutapp2022/MainActivity.java
package team.singularity.scoutapp2022;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.VibrationEffect;
import android.util.Log;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.Button;
import android.widget.ImageButton;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import org.jetbrains.annotations.Nullable;

public final class MainActivity extends AppCompatActivity {
    final String TAG = "MainActivity";
    DatabaseClass database;
    private final BluetoothClass BLUETOOTH;

    {
        BLUETOOTH = new BluetoothClass(this);
    }

    @SuppressLint("SetTextI18n")
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.setContentView(R.layout.activity_main);

        Button newMatchBtn  = findViewById(R.id.newMatchBtn);
        Button pairDevBtn   = findViewById(R.id.pairDevBtn);
        Button viewDataBtn  = findViewById(R.id.viewData);
        ImageButton backBtn = findViewById(R.id.backBtn);
        TextView versionTv  = findViewById(R.id.version);

        backBtn.setVisibility(View.INVISIBLE);

        pairDevBtn.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                ////Util.vibrate(view, VibrationEffect.createPredefined(VibrationEffect.EFFECT_CLICK));
                MainActivity.this.startActivity(new Intent(MainActivity.this.getApplicationContext(), PairDeviceActivity.class));
            }
        });

        viewDataBtn.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                ////Util.vibrate(view, VibrationEffect.createPredefined(VibrationEffect.EFFECT_CLICK));
                startActivity(new Intent(getBaseContext(), ViewDataActivity.class));
            }
        });

        versionTv.setText((CharSequence)"version " + Util.VERSION);

        newMatchBtn.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                ////Util.vibrate(view, VibrationEffect.createPredefined(VibrationEffect.EFFECT_CLICK));
                MainActivity.this.startActivity(new Intent(MainActivity.this.getApplicationContext(), TeamsActivity.class));
            }
        });

        /*(Button)(findViewById(R.id.viewData)).setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                Util.alert(this, "Not implemented!", "Go away!!", new OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        ;
                    }
                })
            }
        });*/
    }

    protected void onStart() {
        super.onStart();
        Log.i(TAG, "Started MainActivity");
        Log.i(TAG, "Setting up bluetooth");
        BLUETOOTH.setup();
        DatabaseClass.setup(BLUETOOTH);
        String data = "{\"teamData\":[],\"matchData\":[]}";
        Log.i(TAG, "Sending data '" + data + '\'');
        BLUETOOTH.send(data);
    }
}

File: app/src/main/java/team/singularity/scoutapp2022/MainActivity.kt
package team.singularity.scoutapp2022

import android.annotation.SuppressLint
import android.bluetooth.BluetoothSocket
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    private val tag = "MainActivity"
    var database: Database? = null
    private var bluetooth: BluetoothClass = BluetoothClass(this)
    var started = false

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.newMatchBtn).setOnClickListener {
            //Log.i(tag, "Clicked new match button")

            val teams = Intent(applicationContext, Teams::class.java)
            startActivity(teams)
        }

        findViewById<TextView>(R.id.version).text = "version $VERSION"
    }

    override fun onStart() {
        super.onStart()
        Log.i(tag, "Started MainActivity")
            Log.i(tag, "Setting up bluetooth")
            bluetooth.setup()
            val data = "{\"teamData\":[],\"matchData\":[]}"
            Log.i(tag, "Sending data '$data'")
            bluetooth.send(data)
    }
}
